up42/asset.py

```diff
@@ -128,12 +126,28 @@ def update_metadata(
         self._info = response_json
         return self._info
 
-    def _get_download_url(self) -> str:
-        url = f"{self.auth._endpoint()}/v2/assets/{self.asset_id}/download-url"
-        response_json = self.auth._request(request_type="POST", url=url)
+    def _get_download_url(self, stac_asset_id: Optional[str] = None, request_type:str = "POST") -> str:
+        if stac_asset_id is None:
+            url = f"{self.auth._endpoint()}/v2/assets/{self.asset_id}/download-url"
+        else:
+            url = f"{self.auth._endpoint()}/v2/assets/{stac_asset_id}/download-url"
+        response_json = self.auth._request(request_type=request_type, url=url)
         download_url = response_json["url"]
         return download_url
 
+    def get_stac_asset_url(self, stac_asset: pystac.Asset):
+        """
+        Returns the signed URL for the STAC Asset.
+        Args:
+            stac_asset: pystac Asset object.
+
+        Returns:
+            Signed URL for the STAC Asset.
```

Comment:
```suggestion
            Pre-signed URL for the STAC Asset.
```
I believe this is technically a pre-signed URL.